### PR TITLE
[ProfileSettings]: Added biometrics in password change view

### LINKED ISSUE #13302: [Settings - Change Password] Add "Enable biometrics option in change password view
### Description
[Settings - Change Password] Add "Enable biometrics option in change password view

bm

### Acceptance Criteria
Enable bio metrics:
https://www.figma.com/file/d0G7m8X6ELjQlFOEKQpn1g/Profile-WIP?type=design&node-id=11-113999&mode=dev

Disable bio metrics:
https://www.figma.com/file/d0G7m8X6ELjQlFOEKQpn1g/Profile-WIP?type=design&node-id=11-115317&mode=dev

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #13302

### What does the PR do
ProfileSettings: Added biometrics in password change view

### Affected areas
ProfileSettings password change view

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/31625338/68a91a5e-5d58-4ac6-80ae-d4f63b07bae5

### CODE REVIEW

File: ui/app/AppLayouts/Profile/views/ChangePasswordView.qml [hunk 1]
```diff
@@ -12,27 +12,82 @@ import StatusQ.Controls 0.1
...
 import AppLayouts.Profile.popups 1.0
 
 SettingsContentBase {
     id: root
 
     property var privacyStore
+
+    readonly property bool biometricsEnabled: localAccountSettings.storeToKeychainValue === Constants.keychain.storedValue.store
+    onBiometricsEnabledChanged: {
+        biometricsSwitch.checked = Qt.binding(() => { return root.biometricsEnabled });
```
Comment (alexjba) [inline ui/app/AppLayouts/Profile/views/ChangePasswordView.qml:26]: The `biometricsSwitch` is in a loader. On the first event the `biometricsSwitch` may not exist. An undefined check would be useful here.

File: ui/app/AppLayouts/Profile/views/ChangePasswordView.qml [hunk 2]
```diff
@@ -12,27 +12,82 @@ import StatusQ.Controls 0.1
...
+            }
+            StatusToolTip {
+                x: 15
+                visible: (!biometricsSwitch.checked && biometricsSwitch.hovered)
+                text: qsTr("Biometric login and transaction authentication")
+            }
+        }
+        StatusModal {
+            id: enableBiometricsPopup
+            title: biometricsSwitch.checked ? qsTr("Disable biometrics") : qsTr("Enable biometrics")
```
Comment (alexjba) [inline ui/app/AppLayouts/Profile/views/ChangePasswordView.qml:51]: I'd link it to the `root.biometricsEnabled` as it's supposed to be the single source of truth for this feature. Looks like the `biometricsSwitch.checked` can have intermediary states. Just to make sure we don't have false positive/negative here.

Same below

File: ui/app/AppLayouts/Profile/views/ChangePasswordView.qml [hunk 3]
```diff
@@ -12,27 +12,82 @@ import StatusQ.Controls 0.1
...
+                    normalColor: "transparent"
+                    onClicked: {
+                        enableBiometricsPopup.close();
+                    }
+                },
+
+                StatusButton {
+                    text: biometricsSwitch.checked ? qsTr("Yes, enable biometrics") : qsTr("Yes, disable biometrics")
+                    onClicked: {
+                        if (biometricsSwitch.checked && !biometricsSwitch.biometricsEnabled) {
```
Comment (alexjba) [inline ui/app/AppLayouts/Profile/views/ChangePasswordView.qml:77]:
```suggestion
                        if (biometricsSwitch.checked && !root.biometricsEnabled) {
```

**Review by alexjba — COMMENTED** [review]
I have a few suggestions below.

Another thing I found is that the buttons are cut when the popup is opened the first time.

Screenshot 2024-02-22 at 10 15 14

File: ui/app/AppLayouts/Profile/views/ChangePasswordView.qml [hunk 4]
```diff
@@ -12,27 +12,82 @@ import StatusQ.Controls 0.1
...
+            LayoutMirroring.enabled: true
+            LayoutMirroring.childrenInherit: true
+            text: qsTr("Enable biometrics")
+            checked: root.biometricsEnabled
+            onToggled: {
+                enableBiometricsPopup.open();
+            }
+            StatusToolTip {
+                x: 15
+                visible: (!biometricsSwitch.checked && biometricsSwitch.hovered)
```
[the remark this answers is not in the document]
Comment (alexandraB99) [reply ui/app/AppLayouts/Profile/views/ChangePasswordView.qml:45]: it's only visible when unchecked https://www.figma.com/file/d0G7m8X6ELjQlFOEKQpn1g/Profile-WIP?type=design&node-id=11-113999&mode=dev

**Comment by alexandraB99** [comment]
> I have a few suggestions below.
> 
> Another thing I found is that the buttons are cut when the popup is opened the first time.
> Screenshot 2024-02-22 at 10 15 14

I cannot reproduce this one, any specific steps?

File: ui/app/AppLayouts/Profile/views/ChangePasswordView.qml [hunk 5]
```diff
@@ -12,27 +12,82 @@ import StatusQ.Controls 0.1
...
-//        implicitHeight: 38
-//        StatusSwitch {
-//            LayoutMirroring.enabled: true
-//            text: qsTr("Enable biometrics")
-//            onToggled: {
-//                //
-//            }
-//        }
-//    }
+    titleRowComponentLoader.sourceComponent: Item {
```
[the remark this answers is not in the document]
Comment (alexandraB99) [reply ui/app/AppLayouts/Profile/views/ChangePasswordView.qml:31]: I was not aware of this requirement, are we sure it's still a thing?

**Comment by alexjba** [comment]
> > I have a few suggestions below.
> > Another thing I found is that the buttons are cut when the popup is opened the first time.
> > Screenshot 2024-02-22 at 10 15 14
> 
> I cannot reproduce this one, any specific steps?

Try restarting the app and opening this popup once. It's happening only on the first open

**Comment by alexandraB99** [comment]
> Also looks wrong when the other popup is asking for password, the Switch is unchecked by itself?

not sure how to react on the password's popup closing signal.

**Comment by alexandraB99** [comment]
> > > I have a few suggestions below.
> > > Another thing I found is that the buttons are cut when the popup is opened the first time.
> > > Screenshot 2024-02-22 at 10 15 14
> > 
> > 
> > I cannot reproduce this one, any specific steps?
> 
> Try restarting the app and opening this popup once. It's happening only on the first open

still can't see it

File: ui/app/AppLayouts/Profile/views/ChangePasswordView.qml (hunk 1 above)
Comment (alexjba) [reply ui/app/AppLayouts/Profile/views/ChangePasswordView.qml:26]: there is a warning when this component gets loaded that the `biometricsSwitch` is undefined. We could remove it this way.

File: ui/app/AppLayouts/Profile/views/ChangePasswordView.qml (hunk 5 above)
[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/AppLayouts/Profile/views/ChangePasswordView.qml:31]: Yes, not supported on Linux

File: ui/app/AppLayouts/Profile/views/ChangePasswordView.qml (hunk 1 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Profile/views/ChangePasswordView.qml]: @alexjba I don't see any warning but I've added the check anyways. in regards to the buttons cut, I can't see that one either

**Review by alexjba — APPROVED** [review]
LGTM! Thanks :D

File: ui/app/AppLayouts/Profile/views/ChangePasswordView.qml [hunk 6]
```diff
@@ -12,27 +12,82 @@ import StatusQ.Controls 0.1
...
+        implicitWidth: 226
+        implicitHeight: 38
+        StatusSwitch {
+            id: biometricsSwitch
+            LayoutMirroring.enabled: true
+            LayoutMirroring.childrenInherit: true
+            visible: Qt.platform.os === Constants.mac
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/views/ChangePasswordView.qml:39]: This should go up to the parent `Item`?

**Review by caybro — CHANGES_REQUESTED** [review]

**Review by caybro — APPROVED** [review]
LGTM :+1:

**Pull request merged**